hyperjumptech/monika
# Fix default maxRedirect too shallow

# Monika Pull Request (PR)  

## What feature/issue does this PR add  
1. Fix issue #1069
2. When probing urls with redirects, monika did not follow and return some error
3. Error MAX redirect from axios was not handled properly
4. Probing with redirect works by default 

## How did you implement / how did you fix it  
1.  Update the default maxRedirect to  3
2. Update docs
3. Handle all possible axios errors

## How to test  
1. Use the same yaml as in the Issue
```yaml
probes:
  - id: 'http-1'
    name: 'status-200-test'
    requests:
      - url: https://www.spotify.com
    alerts:
      - assertion: response.status < 200 or response.status > 2308
        message: HTTP Status is not 200
    incidentThreshold: 3
    recoveryThreshold: 3

notifications:
  - id: desktop
    type: desktop
```
5. Run the probe `npm start -- -c test.yaml`
6. Redirects should work

## before PR
![image](https://github.com/hyperjumptech/monika/assets/964106/d45c32fc-4ee2-474a-ab7a-4f78fa7daafb)

## after PR
![image](https://github.com/hyperjumptech/monika/assets/964106/a8a87d59-4b97-4477-aacf-2b98179fdc52)
